aviatrix/resource_aviatrix_aws_tgw_vpc_attachment.go

```diff
@@ -233,6 +232,17 @@ func resourceAviatrixAwsTgwVpcAttachmentUpdate(d *schema.ResourceData, meta inte
 			return fmt.Errorf("failed to update spoke vpc customized routes: %s", err)
 		}
 	}
+	if d.HasChange("customized_route_advertisement") {
+		awsTgwVpcAttachment := &goaviatrix.AwsTgwVpcAttachment{
+			TgwName:                      d.Get("tgw_name").(string),
+			VpcID:                        d.Get("vpc_id").(string),
+			CustomizedRouteAdvertisement: d.Get("customized_route_advertisement").(string),
+		}
+		err := client.EditTgwSpokeVpcCustomizedRouteAdvertisement(awsTgwVpcAttachment)
+		if err != nil {
+			return fmt.Errorf("failed to update spoke vpc customized routes advertisement: %s", err)
+		}
+	}
 
 	d.Partial(false)
 	return resourceAviatrixAwsTgwVpcAttachmentRead(d, meta)
```

Comment:
Better use the deferred read now that the edit has a chance of failure

Reply:
Will implement defer function for update for all resources in future.